examples/charge_offpeak.py

```diff
@@ -0,0 +1,137 @@
+# -*- coding: utf-8 -*-
+"""Max SOC script.
+
+This script will check the charging level once every minute
+and if the current battery level meets or exceeds the specified
+maximum value, the charging will stop if the vehicle is currently charging.
+
+If the vehicle is charging during off-peak hours, the charging will stop.
+
+Todo: figure out whether the car is plugged in at home or somewhere else.
+
+"""
+
+import jlrpy
+import threading
+import datetime
+import math
+from datetime import date
+import os
+import configparser
+
+
+# login info (email and password) are read from $HOME/.jlrpy.cfg
+# which contains a single line with email and password separated by ':'
+# email@example.com:PassW0rd
+# passwords containing a ':' are not allowed
+
+logger = jlrpy.logger
+
+def distance(origin, destination):
+    """
+    From https://stackoverflow.com/questions/19412462
+
+    Calculate the Haversine distance.
+
+    Parameters
+    ----------
+    origin : tuple of float
+        (lat, long)
+    destination : tuple of float
+        (lat, long)
+
+    Returns
+    -------
+    distance_in_km : float
+
+    Examples
+    --------
+    >>> origin = (48.1372, 11.5756)  # Munich
+    >>> destination = (52.5186, 13.4083)  # Berlin
+    >>> round(distance(origin, destination), 1)
+    504.2
+    """
+    lat1, lon1 = origin
+    lat2, lon2 = destination
+    radius = 6371  # km
+
+    dlat = math.radians(lat2 - lat1)
+    dlon = math.radians(lon2 - lon1)
+    a = (math.sin(dlat / 2) * math.sin(dlat / 2) +
+         math.cos(math.radians(lat1)) * math.cos(math.radians(lat2)) *
+         math.sin(dlon / 2) * math.sin(dlon / 2))
+    c = 2 * math.atan2(math.sqrt(a), math.sqrt(1 - a))
+    d = radius * c
+
+    return d
+
```

Comment:
Too many linebreaks here.

Reply:
Okay. It was a straight copy/paste of code I found which I didn't edit for style. Will remove them.